


package ProtoBuf;


//客户端到服务器协议号
enum ClientToServerCmd
{

	//登录模块
	CLIENT_TO_SERVER_LOGIN_BEGIN			= 1;
	CLIENT_TO_LOGIN_PING					= 2;
	CLIENT_TO_LOGIN_CHECK_VERSION			= 3;
	CLIENT_TO_LOGIN_ACCOUNT_LOGIN			= 4;
	CLIENT_TO_LOGIN_BREAK_LOGIN				= 5;

	CLIENT_TO_LOGIC_PING					= 10;
	CLIENT_TO_LOGIC_RECONNECT				= 11;		// 断线重连
	CLIENT_TO_LOGIC_RESET_TIMEOUT			= 12;		// 重置连接超时时间
	CLIENT_TO_LOGIC_LOGIN_LOGICSERVER		= 13;		// 客户端用账号登录逻辑服务器
	CLIENT_TO_LOGIC_GET_CHARACTER_LIST		= 14;		// 请求角色列表
	CLIENT_TO_LOGIC_CREATE_CHARACTER		= 15;		// 请求创建角色
	CLIENT_TO_LOGIC_DEL_CHARACTER			= 16;		// 请求删除角色
	CLIENT_TO_LOGIC_ENTER_WORLD				= 17;		// 玩家进入游戏世界，与切换场景不同，此时应该服务器应该同步玩家所有数据
	CLIENT_TO_LOGIC_LEAVE_WORLD				= 18;		// 玩家离开游戏
	CLIENT_TO_SERVER_LOGIN_END				= 50;

	//系统模块
	CLIENT_TO_SERVER_SYSTEM_BEGIN			= 60;		//系统模块起始ID
	CLIENT_TO_LOGIC_REDIRECT_CENTERSERVER	= 82;		//客户端发送给中心服的重定向协议（先发给逻辑服，再转给中心服）
	CLIENT_TO_SERVER_SYSTEM_END				= 100;		//系统模块结束ID


	//玩家模块
	CLIENT_TO_SERVER_PLAYER_BEGIN			= 200;
	CLIENT_TO_LOGIC_PLAYER_ATRR_POINT       = 201;   //玩家属性点分配
	CLIENT_TO_LOGIC_PLAYER_ATRR_CLEAR_POINT	= 202;   //属性点洗点
	CLIENT_TO_LOGIC_PLAYER_AUTO_POINT		= 203;   //自动分配属性点
	CLIENT_TO_SERVER_PLAYER_END				= 210;

	//技能模块
	CLIENT_TO_SERVER_SKILL_BEGIN              = 220;      
	CLIENT_TO_LOGIC_SKILL_USE				  = 221;      // 技能使用
	CLIENT_TO_LOGIC_SKILL_CHARGE_END		  = 222;      // 蓄力结束(蓄力技能专用,正常时间内结束不用发)
	CLIENT_TO_LOGIC_SKILL_LEVEL_UP_REQ   	  = 223;      // 技能升级
	CLIENT_TO_LOGIC_SKILL_STOP		= 224	;	//技能中断	
	CLIENT_TO_SERVER_SKILL_END                = 230;      

	//背包模块
	CLIENT_TO_SERVER_PACKAGE_BEGIN				= 240; 
	CLIENT_TO_LOGIC_PACKAGE_INFO				= 241;		//背包物品信息	
	CLIENT_TO_LOGIC_PACKAGE_SORT				= 242;		//整理背包
	CLIENT_TO_LOGIC_PACKAGE_SELL				= 243;		//出售物品
	CLIENT_TO_LOGIC_PACKAGE_USE					= 244;		//使用物品
	CLIENT_TO_LOGIC_EQUIP_STREN					= 245;		//强化
	CLIENT_TO_LOGIC_EQUIP_ONEKEY_STREN			= 246;		//一键强化
	CLIENT_TO_LOGIC_EQUIP_REFINE				= 247;		//精炼
	CLIENT_TO_LOGIC_EQUIP_DIGHOLE				= 248;		//开孔
	CLIENT_TO_LOGIC_EQUIP_INLAY					= 249;		//镶嵌宝石
	CLIENT_TO_LOGIC_EQUIP_TAKEOFF				= 250;		//取下宝石
	CLIENT_TO_LOGIC_EQUIP_COMPOSE				= 251;		//宝石合成
	CLIENT_TO_LOGIC_EQUIP_RECAST				= 252;		//重铸
	CLIENT_TO_LOGIC_EQUIP_INLAY_HOLY			= 253;		//镶嵌神印石
	CLIENT_TO_LOGIC_EQUIP_TAKEOFF_HOLY			= 254;		//取下神印石
	CLIENT_TO_LOGIC_EQUIP_COMPOSE_MATERIAL		= 255;		//材料合成
	CLIENT_TO_LOGIC_PLAYER_EQUIP_INFO			= 256;		//玩家装备信息
	CLIENT_TO_LOGIC_PLAYER_EQUIP_DRESS			= 257;		//穿戴装备
	CLIENT_TO_LOGIC_PLAYER_EQUIP_UNDRESS		= 258;		//卸下装备
	CLIENT_TO_LOGIC_PLAYER_EQUIP_ONEKEY_DRESS	= 259;		//一键穿戴 
	CLIENT_TO_LOGIC_PLAYER_PICKUP_DROP			= 260;		//拾取掉落
	CLIENT_TO_SERVER_PACKAGE_END				= 300; 


	//副本模块
	CLIENT_TO_SERVER_DUPLICATE_BEGIN		= 310;
	CLIENT_TO_LOGIC_DUPLICATE_INFO			= 311;		//副本信息
	CLIENT_TO_LOGIC_DUPLICATE_ENTER			= 312;		//进入副本
	CLIENT_TO_LOGIC_DUPLICATE_EXIT			= 313;		//退出副本
	CLIENT_TO_LOGIC_DUPLICATE_RESULT		= 314;		//副本结果
	CLIENT_TO_LOGIC_DUPLICATE_AUTO_FIGHT	= 315;		//副本自动战斗
	CLIENT_TO_LOGIC_DUPLICATE_LOTTERY		= 316;		//翻牌
	CLIENT_TO_LOGIC_DUPLICATE_BUY_NUM		= 317;		//购买次数
	CLIENT_TO_LOGIC_DUPLICATE_SWEEP			= 318;		//扫荡
	CLIENT_TO_LOGIC_DUPLICATE_RELIVE		= 319;		//复活
	CLIENT_TO_LOGIC_DUPLICATE_LEAVE			= 320;		//离开副本
	CLIENT_TO_LOGIC_DUPLICATE_ENTER_AREA	= 321;		//进入副本区域
	CLIENT_TO_SERVER_DUPLICATE_END			= 350; 


	//移动模块
	CLIENT_TO_SERVER_MOVE_BEGIN             = 360;
	CLIENT_TO_LOGIC_SCENE_REQ_ROUTEINFO     = 361;      // 请求当前地图线路信息
	CLIENT_TO_LOGIC_SCENE_TRANS				= 362;      // 请求场景传送
	CLIENT_TO_LOGIC_SCENE_LOAD_SUCCESS		= 363;		// 地图加载完成
	CLIENT_TO_LOGIC_CHARACTER_SKILLMOVE     = 364;      // 技能移动
	CLIENT_TO_LOGIC_CHARACTER_MOVE			= 365;		// 角色移动
	CLIENT_TO_LOGIC_CHARACTER_ROCKER_MOVE	= 366;		//角色摇杠移动
	CLIENT_TO_LOGIC_CHARACTER_POS_UPDATE	= 367;		//角色位置由客户端同步
	CLIENT_TO_SERVER_MOVE_END               = 390;

	//任务模块
	CLIENT_TO_SERVER_MISSION_BEGIN			= 400;	
	CLIENT_TO_LOGIC_QUERY_MiSSIONLIST		= 401;	//请求任务列表
	CLIENT_TO_LOGIC_ACCEPTMISSION			= 402;  //请求接取任务
	CLIENT_TO_LOGIC_ABANDONMISSION			= 403;  //请求放弃任务
	CLIENT_TO_LOGIC_SUBMITMISSION			= 404;  //请求提交任务
	CLIENT_TO_LOGIC_TALKWITHNPC				= 405;  //请求对话请求
	CLIENT_TO_LOGIC_GETCAMPMISSIONINFO		= 406;	//请求阵营任务
	CLIENT_TO_LOGIC_DICECAMPMISSION			= 407;	//摇筛子
	CLIENT_TO_LOGIC_GETCAMPBOXREWARD		= 408;	//领取阵营宝箱
	CLIENT_TO_SERVER_MISSION_END			= 420;


	//聊天模块
	CLIENT_TO_SERVER_CHAT_BEGIN				= 430;	
	CLIENT_TO_LOGIC_CHATMSGREQ				= 431;	//请求聊天
	CLIENT_TO_CENTER_ASKOFFLINEMSG			= 432;	//请求好友聊天离线数据
	CLIENT_TO_CENTER_INTRVALASKOFFLINEMSG	= 433;	//分段请求聊天离线数据
	CLIENT_TO_CENTER_CHARACTER_DETAIL_DATA	= 434;	//查看玩家详细数据
	CLIENT_TO_SERVER_CHAT_END				= 450;	

	//好友模块
	CLIENT_TO_SERVER_FRIEND_BEGIN		 = 460;	
	CLIENT_TO_CENTER_APPLY_RELATION		 = 461;  //申请添加好友
	CLIENT_TO_CENTER_AGREE_RELATION		 = 462;  //同意添加好友
	CLIENT_TO_CENTER_DELETE_RELATION	 = 463;  //删除好友
	CLIENT_TO_CENTER_RECENT_CHAT_LIST    = 464;  //请求最近联系人
	CLIENT_TO_CENTER_ONEKEY_FRIEND		 = 465;  //好友推荐
	CLIENT_TO_CENTER_RELATION_INFO		 = 466;  //请求好友信息
	CLIENT_TO_CENTER_SEARCH_USER		 = 467;  //搜索好友
	CLIENT_TO_CENTER_GIVE_PHYSCIAL		 = 468;  //赠送体力
	CLIENT_TO_CENTER_AllGIVE_PHYSICAL	 = 469;  //一键赠送体力
	CLIENT_TO_CENTER_GET_PHYSICAL		 = 470;  //领取体力
	CLIENT_TO_CENTER_GET_PHYSICAL_GIFT   = 471;  //领取体力礼包
	CLIENT_TO_CENTER_APPLYLIST			 = 472;  //获取好友申请列表
	CLIENT_TO_CENTER_CLEARAPPLYLIST		 = 473;  //清除好友申请列表
	CLIENT_TO_SERVER_FRIEND_END			 = 500;	


	//场景
	CLIENT_TO_SERVER_SCENE_BEGIN         = 510;

	CLIENT_TO_SERVER_SCENE_END           = 550;

	//邮件模块
	CLIENT_TO_SERVER_MAIL_BEGIN				= 560;
	CLIENT_TO_LOGIC_SENDMAIL				= 561;
	CLIENT_TO_CENTER_GETMAILITEMLIST		= 562;
	CLIENT_TO_CENTER_ASKMAIL				= 563;
	CLIENT_TO_CENTER_DELETEMAIL				= 564;
	CLIENT_TO_CENTER_CHECKNAME				= 565;
	CLIENT_TO_CENTER_GETMAILATTACHMENT		= 566;
	CLIENT_TO_CENTER_GETALLMAILATTACHMENT	= 567;
	CLIENT_TO_CENTER_NDRMAIL				= 568;
	CLIENT_TO_SERVER_MAIL_END				= 600;

	//阵营
	CLIENT_TO_SERVER_CAMP_BEGIN			 = 610;
	CLIENT_TO_CENTER_CAMP_JOIN			 = 611;		//加入阵营
	CLIENT_TO_CENTER_CAMP_CHANGE		 = 612;		//更换阵营
	CLIENT_TO_CENTER_CAMP_INFO			 = 613;		//阵营信息
	CLIENT_TO_CENTER_CAMP_RANK_INFO		 = 614;		//阵营排名信息
	CLIENT_TO_CENTER_CAMP_HONOR_INFO	 = 615;		//阵营荣誉信息
	CLIENT_TO_CENTER_CAMP_PUBLISH_NOTICE = 616;		//发布阵营公告
	CLIENT_TO_SERVER_CAMP_END			 = 650;

	//成就
	CLIENT_TO_SERVER_ACHIEVEMENT_BEGIN		 = 700;
	CLIENT_TO_SERVER_ACHIEVEMENTINFO		 = 701;		//请求成就列表信息
	CLIENT_TO_SERVER_GETACHIEVEMENTREWARD	 = 702;		//领取单个成就奖励
	CLIENT_TO_SERVER_GETACHIEVEMENTBOXREWARD = 703;		//领取成就宝箱
	CLIENT_TO_SERVER_ACHIEVEMENT_END		 = 720;
	

	//副业
	CLIENT_TO_SERVER_DEPUTY_BEGIN					= 730;
	CLIENT_TO_SERVER_DEPUTY_COLLECT_SKILL_INFO		= 731;	//请求收集副业技能信息
	CLIENT_TO_SERVER_DEPUTY_MAKE_SKILL_INFO			= 732;	//请求制造副业技能信息
	CLIENT_TO_SERVER_DEPUTY_UPGRADE_COLLECT_SKILL	= 733;	//请求升级收集副业技能
	CLIENT_TO_SERVER_DEPUTY_REFINE_MAKE_SKILL		= 734;	//请求精修制造副业技能
	CLIENT_TO_SERVER_DEPUTY_COLLECT_OPERATE			= 735;	//请求采集副业操作
	CLIENT_TO_SERVER_DEPUTY_MAKE_OPERATE			= 736;	//请求制作副业操作
	CLIENT_TO_SERVER_DEPUTY_END						= 750;


	//PVP
	CLIENT_TO_SERVER_PVP_BEGIN				= 770;
	CLIENT_TO_SERVER_PVP_SET_MODE			= 771; //设置PK模式
	CLIENT_TO_SERVER_PVP_END				= 779;

	//商城
	CLIENT_TO_SERVER_CENTER_BEGIN			= 780;
	CLIENT_TO_CENTER_GET_MALL_DATA			= 781;	//请求商城数据
	CLIENT_TO_LOGIC_BUY						= 782;  //请求购买
	CLIENT_TO_CENTER_GET_TIME				= 783;	//请求商品剩余刷新时间
	CLIENT_TO_SERVER_CENTER_END				= 800;
};
